packages/react/toggle-button/src/ToggleButton.tsx

```diff
@@ -1,10 +1,199 @@
+  /** The controlled value of the toggled button */
+  value?: string[];
+  /** The uncontrolled value of the toggled button */
+  defaultValue?: string[];
+  /** A function called when the value of the toggled buttons changes */
+  onValueChange?(value: string[]): void;
+};
+
+const ToggleButtonGroup = forwardRefWithAs<typeof GROUP_DEFAULT_TAG, ToggleButtonGroupOwnProps>(
+  (props, forwardedRef) => {
+    const {
+      as: Comp = GROUP_DEFAULT_TAG,
+      value: valueProp,
+      defaultValue,
+      onValueChange,
+      children,
+      selectionIsRequired = false,
+      rovingFocus = false,
+      ...groupProps
+    } = props;
+
+    const [value, setValue] = useControlledState<string[]>({
+      prop: valueProp,
+      defaultProp: defaultValue || [],
+      onChange: onValueChange,
+      isEqual: stringArraysAreEqual,
+    });
+
+    function handleChange(buttonValue: string) {
+      if (!value || value.length < 1) {
+        return setValue([buttonValue]);
+      }
+
+      if (selectionIsRequired && value.length === 1 && value.includes(buttonValue)) {
+        return;
+      }
+
+      setValue(
+        value.includes(buttonValue)
+          ? value.filter((v) => v !== buttonValue)
+          : value.concat(buttonValue).sort()
+      );
+    }
+
+    const context: ToggleButtonGroupContextValue = {
+      setValue,
+      value,
+      handleChange,
+      rovingFocus,
+      name: GROUP_CONTEXT_NAME,
+    };
+
+    return (
+      <Comp {...getPartDataAttrObj(GROUP_NAME)} role="group" ref={forwardedRef} {...groupProps}>
+        <ToggleButtonGroupContext.Provider value={context}>
+          <ToggleButtonGroupInner rovingFocus={rovingFocus}>{children}</ToggleButtonGroupInner>
+        </ToggleButtonGroupContext.Provider>
+      </Comp>
+    );
+  }
+);
+
+function ToggleButtonGroupInner(props: { rovingFocus: boolean; children: React.ReactNode }) {
+  return props.rovingFocus ? (
+    <RovingFocusGroup loop>{props.children}</RovingFocusGroup>
+  ) : (
+    <React.Fragment>{props.children}</React.Fragment>
+  );
+}
+
+/* -------------------------------------------------------------------------------------------------
+ * ToggleButtonExclusive
+ * -----------------------------------------------------------------------------------------------*/
+
+const GROUP_EXC_NAME = 'ToggleButtonGroupExclusive';
+const GROUP_EXC_CONTEXT_NAME = 'ToggleButtonGroupExclusiveContext';
+
+type ToggleButtonGroupExclusiveOwnProps = ToggleButtonGroupSharedProps & {
+  /** The controlled value of the toggled button */
+  value?: string | null;
+  /** The uncontrolled value of the toggled button */
+  defaultValue?: string;
+  /** A function called when the value of the toggled buttons changes */
+  onValueChange?(value: string | null): void;
+};
+
+type ToggleButtonGroupExclusiveContextValue = {
+  value: string | null;
+  setValue:
+    | React.Dispatch<React.SetStateAction<string | null>>
+    | React.Dispatch<React.SetStateAction<string>>;
```

Comment:
Why the two types here? Looks like the first makes the second redundant?

Reply:
Because if a consumer declared `const [state, setState] = React.useState('value')` they'll get an error from passing `setState` to `setValue` if we don't allow the second type. I ran into this in one of the stories.

Reply:
Sorry, that's incorrect. I think this was an issue internally I was accounting for but it doesn't seem to be an issue now, so I removed the second type.